app/assets/javascripts/angular/resources/shared_graph_behavior.js

```diff
@@ -60,6 +60,22 @@ angular.module("Prometheus.services").factory("SharedGraphBehavior", ["$http", "
       }
     }
 
+    function addVarsFromServerInterpolations(servers) {
+      var singleInterpolation = /{{\s?(\w+)\s?}}/g;
+      for (var i = 0; i < servers.length; i++) {
+        while (match = singleInterpolation.exec(servers[i].url)) {
+          var_name = match[1]
+          if (!(var_name in $scope.globalConfig.vars)) {
+            console.log("Found " + var_name + " in " + servers[i].url)
+            $scope.globalConfig.vars[var_name] = ''  // no default value
+          }
+        }
+      }
+    }
+
+    $scope.servers = servers;
```

Comment:
it would be cool to not have the unnecessary `while`, i think it's potentially misleading because it doesn't actually loop in this case. `var_name` was also an unintentional global, and since it is so short lived I prefer single character variable names.

```javascript
function addVarsFromServerInterpolations(servers) {
  var re = /{{\s?(\w+)\s?}}/g;
  for (var i = 0; i < servers.length; i++) {
    var m = re.exec(servers[i].url);
    if (m && !$scope.globalConfig.vars[m[1]]) {
      $scope.globalConfig.vars[m[1]] = '';
    }
  }
}
```